lib/frodata/concerns/api.rb

```diff
@@ -0,0 +1,262 @@
+# frozen_string_literal: true
+
+require 'erb'
+require 'uri'
+require 'frodata/concerns/verbs'
+
+module FrOData
+  module Concerns
+    module API
+      extend FrOData::Concerns::Verbs
+
+      # Public: Helper methods for performing arbitrary actions against the API using
+      # various HTTP verbs.
+      #
+      # Examples
+      #
+      #   # Perform a get request
+      #   client.get '/api/data/v9.1/leads'
+      #   client.api_get 'leads'
+      #
+      #   # Perform a post request
+      #   client.post '/api/data/v9.1/leads', { ... }
+      #   client.api_post 'leads', { ... }
+      #
+      #   # Perform a put request
+      #   client.put '/api/data/v9.1/leads(073ca9c8-2a41-e911-a81d-000d3a1d5a0b)', { ... }
+      #   client.api_put 'leads(073ca9c8-2a41-e911-a81d-000d3a1d5a0b)', { ... }
+      #
+      #   # Perform a delete request
+      #   client.delete '/api/data/v9.1/leads(073ca9c8-2a41-e911-a81d-000d3a1d5a0b)'
+      #   client.api_delete 'leads(073ca9c8-2a41-e911-a81d-000d3a1d5a0b)'
+      #
+      # Returns the Faraday::Response.
+      define_verbs :get, :post, :put, :delete, :patch, :head
+
+      def metadata
+        api_get("$metadata").body
+      end
+
+      # Public: Executs a SOQL query and returns the result.
+      #
+      # soql - A SOQL expression.
+      #
+      # Examples
+      #
+      #   # Find the names of all Accounts
+      #   client.query('select Name from Account').map(&:Name)
+      #   # => ['Foo Bar Inc.', 'Whizbang Corp']
+      #
+      # Returns a Restforce::Collection if Restforce.configuration.mashify is true.
+      # Returns an Array of Hash for each record in the result if
+      # Restforce.configuration.mashify is false.
+      def query(soql)
+        response = api_get 'query', q: soql
+        mashify? ? response.body : response.body['records']
+      end
+
+      # Public: Insert a new record.
+      #
+      # entity_set - The set the entity belongs to
+      # attrs   - Hash of attributes to set on the new record.
+      #
+      # Examples
+      #
+      #   # Add a new lead
+      #   client.create('leads', {"firstname" =>'Bob'})
+      #   # => '073ca9c8-2a41-e911-a81d-000d3a1d5a0b'
+      #
+      # Returns the String Guid of the newly created entity.
```

Comment:
It doesn't have to return only a guid? It could just be whatever the key is correct?

Reply:
As an example the  key could be an integer.

Reply:
yea I ll mmake it more generic . message